### ISSUE TITLE
Crash report 4.7.1: ArrayIndexOutOfBoundsException in ReaderPostService

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#3478]
```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=12; index=-1
       at java.util.ArrayList.get(ArrayList.java:310)
       at org.wordpress.android.ui.reader.services.ReaderPostService$10.run(ReaderPostService.java:307)
```

**Comment by nbradbury** [comment]
Note: 73 instances of this, all of which show `Fatal Exception: java.lang.ArrayIndexOutOfBoundsException, length=12; index=-1`

**Comment by tonyr59h** [comment]
Addressed in #3482.
